onnx_chainer/functions/array.py

```diff
@@ -52,10 +52,85 @@ def convert_Depth2Space(func, onnx_op_name, opset_version, input_names,
 
 def convert_GetItem(func, onnx_op_name, opset_version, input_names,
                     output_names, parameters):
-    return helper.make_node(
-        onnx_op_name, input_names, output_names,
-        slice=func.slices
-    ),
+    x = func.inputs[0]
+    axes, starts, ends = [], [], []
+    squeeze_idxs, unsqueeze_idxs = [], []
+    skipped = 0  # when set ellipsis, need to skip index rolling
+
+    for i, idx in enumerate(func.slices):
+        # axis means the index of input x, adjust None and Ellipsis counts
+        axis = i - len(unsqueeze_idxs) + skipped
+        if isinstance(idx, slice):
+            if idx.step is not None and idx.step != 1:
+                raise ValueError(
+                    'GetItem with {}step slicing is not supported in ONNX '
+                    'Slice operator'.format(idx.step))
+            axes.append(axis)
+            starts.append(idx.start)
+            ends.append(idx.stop)
+        elif isinstance(idx, int):
+            axes.append(axis)
+            starts.append(idx)
+            ends.append(idx+1)
+            squeeze_idxs.append(axis)
+        elif idx is None:
+            unsqueeze_idxs.append(i - len(squeeze_idxs) + skipped)
+        elif idx is Ellipsis:
+            # calculate rest slice number except None, GetItem does not allow
+            # multiple Ellipsis, so ignore latter Ellipsis count
+            rest_slice_len = len(
+                [idx_ for idx_ in func.slices[i+1:] if idx_ is not None])
+            for ellipsis_i in range(axis, len(x.shape)-rest_slice_len):
```

Comment:
I'm not sure, but is this loop actually necessary?

```
assert skiped == 0
skipped = len(x.shape) - rest_slice_len - 1
```
or something would work?

posted in proxy, https://github.com/chainer/onnx-chainer/pull/72#discussion_r254584417

Reply:
In this loop takes the following case.

```
>>> in_shape = (3,3,3,3)
>>> x = np.zeros(in_shape)
>>> y = F.get_item(x, (Ellipsis, slice(1, None)))
>>> # y = x[..., 1:]  # equals
```

In such a case, the converter need to setup slicing information about axis 0~2, the loop makes the below variables.

```
axes = [0, 1, 2]
starts = [0, 0, 0]
ends = [3, 3, 3]
```

And next loop, the parent converter appends axis=3, start=1, ends=3.

This loop looks ugly and I afraid it will be written more simple, however. @shinh

Reply:
Hmm sorry, but I still don't understand. I thought my suggested change produces Slice with `axes=[3], starts=[1], ends[3]` for your code. Shouldn't it be OK?

Reply:
I see, I misunderstood axes attribute accepts only `null` or `[0, ..., ndim-1]`, but axes is looked up only slicing index. thx!